httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/nio/ClientH2StreamHandler.java

```diff
@@ -166,21 +166,18 @@ private void commitRequest(final HttpRequest request, final EntityDetails entity
 
     @Override
     public void produceOutput() throws HttpException, IOException {
-        switch (requestState) {
-            case HEADERS:
-                exchangeHandler.produceRequest(new RequestChannel() {
+        if (requestState == MessageState.HEADERS) {
```

Comment:
@arturobernalg Why? What is exactly wrong with `switch case` here? How is this an improvement?

Reply:
Hi @ok2c. 

IMO Increase the code readability. If you notice I only change those statement with only one or two conditions.

Reply:
@arturobernalg I do not see exactly how this makes anything better readable. This is _very_ subjective and purely driven by personal fancy of Sonar developers.